packet.go

```diff
@@ -177,6 +241,11 @@ func (p *Packet) decodeIp(recur int) {
 	if p.iphdr == nil {
 		p.iphdr = new(Iphdr)
 	}
+	if recur > 0 {
```

Comment:
IP IP does not have mac right? only gre.

Reply:
ip-ip and gre both doesn't have mac. gre uses EthProto to identify next protocol. It is same Ethproto used by ethernet header to identify next protocol.